lang/py/avro/test/test_tether_word_count.py

```diff
@@ -63,6 +64,23 @@ def _version():
                                 os.path.dirname(__file__)])
 
 
+def _has_java():
+  """Detect if this system has a usable java installed.
+
+  On most systems, this is just checking if `java` is in the PATH.
+
+  But macos always has a /usr/bin/java, which does not mean java is installed. If you invoke java on macos and java is not installed, macos will spawn a popup telling you how to install java. This code does additional work around that to be completely automatic.
+  """
+  if platform.system() == "Darwin":
+    try:
+      output = subprocess.check_output("/usr/libexec/java_home", stderr=subprocess.STDOUT)
+    except subprocess.CalledProcessError as e:
+      output = e.output
+    return ("No Java runtime present" not in output)
+  return bool(distutils.spawn.find_executable("java"))
+
+
+@unittest.skipUnless(_has_java(), "No Java runtime present")
```

Comment:
To clarify my comment: you might also want to skip this test if `_JAR_PATH` doesn't exist.  The current behaviour in the build.xml is to use ivy to *ensure* that it exists (by downloading from apache snapshots if necessary).